src/session.cpp

```diff
@@ -94,11 +100,11 @@ std::size_t olm::Session::new_outbound_session(
     std::uint8_t secret[4 * CURVE25519_SHARED_SECRET_LENGTH];
     std::uint8_t * pos = secret;
 
-    _olm_crypto_curve25519_shared_secret(&alice_identity_key_pair, &one_time_key, pos);
+    _olm_crypto_curve25519_shared_secret(&alice_identity_key_pair, &bob_one_time_key, pos);
     pos += CURVE25519_SHARED_SECRET_LENGTH;
     _olm_crypto_curve25519_shared_secret(&base_key, &identity_key, pos);
     pos += CURVE25519_SHARED_SECRET_LENGTH;
-    _olm_crypto_curve25519_shared_secret(&base_key, &one_time_key, pos);
+    _olm_crypto_curve25519_shared_secret(&base_key, &bob_one_time_key, pos );
     pos += CURVE25519_SHARED_SECRET_LENGTH;
     _olm_crypto_curve25519_shared_secret(&base_key, &pre_key, pos);
```

Comment:
The initial plan on Linear was to follow our existing fork of Olm if there is an OTK (4 DH steps), but to follow the Signal spec if there is no OTK. This is what I ran by @yiannisdt.

Here, you're doing something slightly differently. If there is no OTK, you still run 4 DH steps. But the 3rd and 4th are the exact same.

I'm not sure if this is cryptographically safe. I think it would be safest to stick with the original plan, but we can also check with @yiannisdt in a second about it, since we have a meeting coming up.

Reply:
Good call with that!

I update it to avoid doing the same step twice and stick to 3 steps when there is no OTK.